### PR TITLE
don't use CONAN_PKG for diligent-core

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **diligent-core/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/diligent-core/all/CMakeLists.txt [hunk 1]
```diff
@@ -4,6 +4,8 @@ project(cmake_wrapper)
 include(conanbuildinfo.cmake)
 conan_basic_setup(TARGETS)
 
+find_package(spirv-tools CONFIG)
+find_package(spirv-cross CONFIG)
```
Comment (SpaceIm) [inline recipes/diligent-core/all/CMakeLists.txt:8]:
```suggestion
find_package(SPIRV-Tools REQUIRED CONFIG)
find_package(spirv-cross REQUIRED CONFIG)
```

File: recipes/diligent-core/all/patches/0002-use_conan_dependencies.patch [hunk 2]
```diff
@@ -20,7 +20,7 @@ index 00e634263..a7519f23e 100644
      target_link_libraries(Diligent-ShaderTools
      PRIVATE
 -        spirv-cross-core
-+        CONAN_PKG::spirv-cross
++        spirv-cross::spirv-cross-core
```
Comment (SpaceIm) [inline recipes/diligent-core/all/patches/0002-use_conan_dependencies.patch:23]:
```suggestion
+        spirv-cross::spirv-cross-core
```

no, `spirv-cross-core` is the correct target

File: recipes/diligent-core/all/patches/0002-use_conan_dependencies.patch [hunk 3]
```diff
@@ -119,7 +119,7 @@ index e329495f4..f2adeccdd 100644
 -            target_compile_options(${TARGET} PRIVATE "$<$<CONFIG:DEBUG>:/Ob1>")
 -        endforeach()
 -    endif()
-+    list(APPEND THIRD_PARTY_LIBS spirv-tools::SPIRV-Tools-opt CONAN_PKG::spirv-cross glslang::SPIRV glslang::glslang glslang::MachineIndependent glslang::GenericCodeGen glslang::OSDependent glslang::OGLCompiler glslang::HLSL)
++    list(APPEND THIRD_PARTY_LIBS spirv-tools::SPIRV-Tools-opt spirv-cross::spirv-cross-msl spirv-cross::spirv-cross-glsl glslang::SPIRV glslang::glslang glslang::MachineIndependent glslang::GenericCodeGen glslang::OSDependent glslang::OGLCompiler glslang::HLSL)
```
Comment (SpaceIm) [inline recipes/diligent-core/all/patches/0002-use_conan_dependencies.patch:122]:
```suggestion
+    list(APPEND THIRD_PARTY_LIBS SPIRV-Tools-opt spirv-cross-msl spirv-cross-glsl glslang::SPIRV glslang::glslang glslang::MachineIndependent glslang::GenericCodeGen glslang::OSDependent glslang::OGLCompiler glslang::HLSL)
```

File: recipes/diligent-core/all/patches/0008-API250014-use_conan_dependencies_in_shader_tools.patch [hunk 4]
```diff
@@ -7,7 +7,7 @@ index e4772d5ea..34451f2dd 100644
      message("HLSL support is disabled. Vulkan backend will not be able to consume SPIRV bytecode generated from HLSL.")
  else()
 -    list(APPEND PRIVATE_DEPENDENCIES SPIRV-Tools-opt)
-+    list(APPEND PRIVATE_DEPENDENCIES CONAN_PKG::spirv-tools)
++    list(APPEND PRIVATE_DEPENDENCIES spirv-tools::SPIRV-Tools-opt)
```
Comment (SpaceIm) [inline recipes/diligent-core/all/patches/0008-API250014-use_conan_dependencies_in_shader_tools.patch:10]: no, `SPIRV-Tools-opt`  is the correct target

File: recipes/diligent-core/all/patches/0008-API250014-use_conan_dependencies_in_shader_tools.patch [hunk 5]
```diff
@@ -20,14 +20,14 @@ index dc47ac4b2..993fe97ab 100644
      target_link_libraries(Diligent-ShaderTools
      PRIVATE
 -        spirv-cross-core
-+        CONAN_PKG::spirv-cross
++        spirv-cross::spirv-cross-core
```
Comment (SpaceIm) [inline recipes/diligent-core/all/patches/0008-API250014-use_conan_dependencies_in_shader_tools.patch:23]: again, no patch here, `spirv-cross-core` is correct

File: recipes/diligent-core/all/patches/0008-API250014-use_conan_dependencies_in_shader_tools.patch [hunk 6]
```diff
@@ -20,14 +20,14 @@ index dc47ac4b2..993fe97ab 100644
...
      )
  
      if (${USE_SPIRV_TOOLS})
          target_link_libraries(Diligent-ShaderTools
          PRIVATE
 -            SPIRV-Tools-opt
-+            CONAN_PKG::spirv-tools
++            spirv-tools::SPIRV-Tools-opt
```
Comment (SpaceIm) [inline recipes/diligent-core/all/patches/0008-API250014-use_conan_dependencies_in_shader_tools.patch:30]: same here, no patch needed

File: recipes/diligent-core/all/patches/0009-API250014-use_conan_dependencies_in_third_party.patch [hunk 7]
```diff
@@ -71,7 +71,7 @@ index d7eea08a..e457b9b6 100644
 -        endforeach()
 -    endif()
 -
-+    list(APPEND THIRD_PARTY_LIBS spirv-tools::SPIRV-Tools-opt CONAN_PKG::spirv-cross glslang::SPIRV glslang::glslang glslang::MachineIndependent glslang::GenericCodeGen glslang::OSDependent glslang::OGLCompiler glslang::HLSL xxhash::xxhash)
++    list(APPEND THIRD_PARTY_LIBS spirv-tools::SPIRV-Tools-opt spirv-cross::spirv-cross-msl spirv-cross::spirv-cross-glsl glslang::SPIRV glslang::glslang glslang::MachineIndependent glslang::GenericCodeGen glslang::OSDependent glslang::OGLCompiler glslang::HLSL xxhash::xxhash)
```
Comment (SpaceIm) [inline recipes/diligent-core/all/patches/0009-API250014-use_conan_dependencies_in_third_party.patch:74]:
```suggestion
+    list(APPEND THIRD_PARTY_LIBS SPIRV-Tools-opt spirv-cross-msl spirv-cross-glsl glslang::SPIRV glslang::glslang glslang::MachineIndependent glslang::GenericCodeGen glslang::OSDependent glslang::OGLCompiler glslang::HLSL xxHash::xxhash)
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

**Comment by AndreyMlashkin** [comment]
Please approve
@lasote
@danimtb
@jgsogo
@czoido
@sse4
@uilianries

File: recipes/diligent-core/all/patches/0002-use_conan_dependencies.patch [hunk 8]
```diff
@@ -1,36 +1,8 @@
...
-index 00e634263..a7519f23e 100644
+index 00e63426..ad9041d9 100644
 --- a/Graphics/ShaderTools/CMakeLists.txt
 +++ b/Graphics/ShaderTools/CMakeLists.txt
-@@ -133,26 +133,26 @@ endif()
- if(ENABLE_SPIRV)
-     target_link_libraries(Diligent-ShaderTools
-     PRIVATE
--        spirv-cross-core
-+        CONAN_PKG::spirv-cross
```
Comment (uilianries) [inline recipes/diligent-core/all/patches/0002-use_conan_dependencies.patch:23]: Avoid CONAN_PKG, it won't be supported on Conan 2.0. Prefer `cmake_find_package` generator in this case.

Comment (AndreyMlashkin) [reply recipes/diligent-core/all/patches/0002-use_conan_dependencies.patch:23]: This line is deleted, you should take first minus in considiration.
Github highlighting of a patch of a patch is obviously wrong

**Review by jgsogo — APPROVED** [review]

File: recipes/diligent-core/all/patches/0002-use_conan_dependencies.patch (hunk 8 above)
Comment (uilianries) [reply recipes/diligent-core/all/patches/0002-use_conan_dependencies.patch:23]: okay, thank you for clarifying

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**